### ISSUE TITLE
Any working upgrade procedure?!?

### ISSUE THREAD

**Issue by MarcoV-git** [ikke-t/awx_pod#16]
Ikke,

Did you ever succeeded in upgrading your install from AWX 9.2.0 towards AWX 10.0.0 or 11.2.0?
I performed several ways to upgrade while retaining the content, but failed always. The things I tried:
- used the database upgrade procedure within the postgres10 image (POSTGRESQL_UPGRADE variable)
- used the database upgrade method used by AWX  (https://github.com/ansible/awx/blob/devel/installer/roles/local_docker/tasks/upgrade_postgres.yml)
- used the database upgrade method from tianon (https://hub.docker.com/r/tianon/postgres-upgrade)
- used postgres 9.6 pg_dump and import in postgres 10.0 AWX setup

but all without success. Most of the times I ran into postgres superuser issues and with the 'successfull' migrations the resulting AWX was empty (no projects, templates, inventories, etc).
I'm kind of surprised that even 'manual' migration of data did not resulted in a successful upgrade..
What are your experiences in trying to upgrade an existing 9.2.0 install..?!?

**Comment by ikke-t** [comment]
No, I actually haven't used awx for a while myself now. I have a Tower instance ATM that I use. As I actually want to upgrade :smile: . I have the data left from home install, I could see if I can upgrade it at some point. It's been on todo list for loooooong time.

**Comment by Klaas-** [comment]
Hi,
https://github.com/ansible/awx/blob/devel/installer/roles/local_docker/tasks/upgrade_postgres.yml that _should_ work. I've sucessfully updated awx from 6.something and 9.3. to 10, but I have been using the sclorg postgres image from start (which includes an upgrade path). I've had to do some manual work on db upgrade though, problem was that the postgres db never shuts down cleanly which sclorg update expects, so I had to manually stop the postgresdb cleanly before upgrading to 10 from 9.6

**Comment by Klaas-** [comment]
I did not find a way to easily migrate from docker library postgres to sclorg postgres images :)

**Comment by MarcoV-git** [comment]
OK, but if I change the database back from sclorg database image to docker library postgres and then use the upgrade_postgres.yml again to manually upgrade the database to v10.0.0 that database should come up with the docker library postgres. And all should be fine. The only thing is that in next upgrades i still need to perform DB upgrade via this manual procedure...
